File: common/src/main/java/wile/redstonepen/platform/INetworkingClientPlatform.java
package wile.redstonepen.platform;

import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
public interface INetworkingClientPlatform
{
  void registerClientReceiver();
  void sendToServer(wile.redstonepen.net.Networking.UnifiedPayload payload);
}

File: common/src/main/kotlin/wile/redstonepen/platform/INetworkingClientPlatform.kt
package wile.redstonepen.platform

import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import wile.redstonepen.net.Networking.UnifiedPayload

@Environment(EnvType.CLIENT)
interface INetworkingClientPlatform {
    fun registerClientReceiver()

    fun sendToServer(payload: UnifiedPayload)
}
